File: BankingApp_WDP.APP/app/src/main/java/com/example/bankingapp/data/model/transaction/DepositRequest.kt
package com.example.bankingapp.data.model.transaction

import com.google.gson.annotations.SerializedName

data class DepositRequest(
    @SerializedName("toAccountId") val toAccountId: String,
    @SerializedName("amount")      val amount: Double,
    @SerializedName("currency")    val currency: String = "MDL",
    @SerializedName("description") val description: String? = null
)

File: BankingApp_WDP.API/DTOs/Transactions/DepositRequest.cs
using System.ComponentModel.DataAnnotations;

namespace BankingApp.DTOs.Transactions;

public class DepositRequest
{
    [Required]
    public int ToAccountId { get; set; }

    [Required, Range(0.01, double.MaxValue, ErrorMessage = "Suma trebuie să fie pozitivă.")]
    public decimal Amount { get; set; }

    [StringLength(3, MinimumLength = 3)]
    public string Currency { get; set; } = "MDL";

    [StringLength(500)]
    public string? Description { get; set; }
}
